cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -522,3 +535,80 @@ func (c *CStorVolumeReplicaController) syncCvr(cvr *apis.CStorVolumeReplica) {
 		cvr.Status.Capacity = *capacity
 	}
 }
+
+func (c *CStorVolumeReplicaController) upgrade(cvr *apis.CStorVolumeReplica) (
+	*apis.CStorVolumeReplica, error,
+) {
+	if cvr.VersionDetails.Current != cvr.VersionDetails.Desired {
+		if !isCurrentVersionValid(cvr) {
+			return nil, pkg_errors.Errorf("invalid current version %s", cvr.VersionDetails.Current)
+		}
+		if !isDesiredVersionValid(cvr) {
+			return nil, pkg_errors.Errorf("invalid desired version %s", cvr.VersionDetails.Desired)
+		}
+		replicaID, err := hash.Hash(cvr.UID)
```

Comment:
this is not valid if some user wants to upgrade to just 1.2 but not 1.3

Reply:
will this be possible to handle in the code where these are used? cc: @kmova 
so, should we consider moving to desired state as the image version?

Reply:
we also need to make sure that - this doesn't gets executed for 1.3 to any later version upgrades